opnode/rollup/driver/step.go

```diff
@@ -84,18 +89,18 @@ func (d *outputImpl) newBlock(ctx context.Context, l2Finalized eth.BlockID, l2Pa
 		BatchV1: derive.BatchV1{
 			Epoch:        rollup.Epoch(l1Info.NumberU64()),
 			Timestamp:    uint64(payload.Timestamp),
-			Transactions: payload.Transactions[depositStart:],
+			Transactions: payload.TransactionsField[depositStart:],
 		},
 	}
-
-	return payload.ID(), batch, nil
+	ref, err := derive.BlockReferences(payload, &d.Config.Genesis)
+	return ref, batch, err
 }
 
 // DriverStep derives and processes one or more L2 blocks from the given sequencing window of L1 blocks.
 // An incomplete sequencing window will result in an incomplete L2 chain if so.
 //
 // After the step completes it returns the block ID of the last processed L2 block, even if an error occurs.
-func (d *outputImpl) step(ctx context.Context, l2Head eth.BlockID, l2Finalized eth.BlockID, unsafeL2Head eth.BlockID, l1Input []eth.BlockID) (out eth.BlockID, err error) {
+func (d *outputImpl) step(ctx context.Context, l2Head eth.L2BlockRef, l2Finalized eth.BlockID, unsafeL2Head eth.BlockID, l1Input []eth.BlockID) (out eth.L2BlockRef, err error) {
```

Comment:
Shouldn't we be removing `l1Input` now that the `l2Head` has L1 block ID info? What's the point of changing to `BlockRef` without using the extra info?

Reply:
This is for timestamps later.

We could also be checking that `l1Input[0].Parent.Hash == l2Head.L1Origin.Hash` as this function advances full epochs at a time.